pyflagser/flagio.py

```diff
@@ -1,84 +1,261 @@
+        `fmt`
+        Matrix representation of a directed/undirected weighted graph. Diagonal
+        elements are vertex weights.
+
+    Notes
+    -----
+    The input graphs cannot contain self-loops, i.e. edges that start and end
+    in the same vertex, therefore diagonal elements of the input adjacency
+    matrix store vertex weights.
+
+    References
+    ----------
+    .. [1] D. Luetgehetmann, "Documentation of the C++ flagser library";
+           `GitHub: <https://github.com/luetge/flagser/blob/master/docs/\
+           documentation_flagser.pdf>`_.
 
     """
+    # Warn if dtype is bool
+    if np.issubdtype(dtype, np.bool_):
+        warnings.warn("dtype is a bool type, you may want to use "
+                      " the load_unweighted_flag function instead.")
+
+    # Handle default parameter
+    if infinity_value is None:
+        if fmt != 'dense':
+            _infinity_value = None
+        else:
+            # Get the maximum value depending on adjacency_matrix.dtype
+            if np.issubdtype(dtype, np.integer):
+                _infinity_value = np.iinfo(dtype).max
+            elif np.issubdtype(dtype, np.float_):
+                _infinity_value = np.inf
+            else:
+                _infinity_value = 0
+    else:
+        if fmt != 'dense':
+            warnings.warn("infinity_value has been specified with a fmt that "
+                          "is not 'dense' and will be ignored.")
+            _infinity_value = None
+        else:
+            _infinity_value = infinity_value
+
     with open(fname, 'r') as f:
         next(f)
         line = f.readline().strip()
-        vertices = list(map(float, line.split(' ')))
-        flag_matrix = sp.csr_matrix((len(vertices), len(vertices)),
-                                    dtype=dtype)
-        flag_matrix.setdiag(vertices)
+        vertices = np.array(line.split(' '), dtype=dtype)
+
+        if fmt == 'dense':
+            adjacency_matrix = np.asarray(
+                _infinity_value
+                * np.ones((len(vertices), len(vertices))), dtype=dtype)
+            adjacency_matrix[np.eye(adjacency_matrix.shape, dtype=np.bool)] = \
+                vertices
+        else:
+            adjacency_matrix = sp.csr_matrix((len(vertices), len(vertices)),
+                                             dtype=dtype)
 
-        for line in f.readlines()[1:]:
-            edge = line.strip().split(' ')
-            flag_matrix[int(float(edge[0])), int(float(edge[1]))] = \
-                float(edge[2])
+        # Silence sparse warnings
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore', sp.SparseEfficiencyWarning)
+            adjacency_matrix.setdiag(vertices)
 
-    return flag_matrix.asformat(fmt)
+            for line in f.readlines()[1:]:
+                edge = line.strip().split(' ')
+                adjacency_matrix[int(float(edge[0])), int(float(edge[1]))] = \
```

Comment:
Same comment/question on the use of `float` as in the unweighted case, for the LHS. By contrast, its use on the RHS here is clear.